src/inmanta/execute/proxy.py

```diff
@@ -134,7 +136,12 @@ def return_value(cls, value: object) -> Union[None, str, tuple, int, float, bool
 
     def __getattr__(self, attribute):
         instance = self._get_instance()
-        value = instance.get_attribute(attribute).get_value()
+
+        try:
+            value = instance.get_attribute(attribute).get_value()
+        except NotFoundException:
+            # allow for hasattr(proxy, "some_attr")
+            raise AttributeError()
```

Comment:
@wouterdb I'd like to explicitly draw your attention to this since it was not strictly part of the ticket. I encountered this issue while writing the test for serializing an entity instance. `hasattr` works by "calling `getattr` and seeing whether it raises an `AttributeError` or not". So I thought it would make sense to add support for that by proxying the exception.

Reply:
This has the potential of breaking modules all over the place (different exception, .... )

Perhaps it is best to put this in a MR of its own, so it is easier to roll back